src/ConnectionPool.jl

```diff
@@ -641,12 +641,15 @@ function getconnection(::Type{TCPSocket},
                        port::AbstractString;
                        keepalive::Bool=false,
                        connect_timeout::Int=0,
+                       readtimeout::Int=0,
                        kw...)::TCPSocket
 
     p::UInt = isempty(port) ? UInt(80) : parse(UInt, port)
 
     @debug 2 "TCP connect: $host:$p..."
 
+    timeouts = filter(!iszero, [connect_timeout, readtimeout])
+    connect_timeout = isempty(timeouts) ? 0 : minimum(timeouts)
```

Comment:
Couldn't this be just
```suggestion
    connect_timeout = min(connect_timeout, readtimeout)
```

Reply:
No, because we want the smallest non-zero between the two. It could be:

```julia
connect_timeout = minimum(filter(!iszero, [connect_timeout, readtimeout]))
```

Reply:
I don't believe so. If either `connect_timeout` or `readtimeout` is 0, then that will return 0, meaning no connection timeout will be set.

Reply:
How about just:
```julia
connect_timeout = connect_timeout == 0 && readtimeout > 0 ? readtimeout : connect_timeout
```
That avoids allocating an unnecessary array and the need to call filter and minimum.

Reply:
Nice, I like that better, fixed.